File: dev-docs/snippets/model-surface/query-one.swift
import JsBaoClient

// Fetch just the first match (with an optional sort). `queryOne` mirrors the
// JS client and resolves to `nil` when nothing matches.
func queryOne() {
  // #region example
  let topTask = Task.queryOne(
    ["completed": false],
    options: QueryOptions(sort: ["priority": -1])
  )
  // #endregion example
  _ = topTask
}

File: dev-docs/snippets/model-surface/query-one.ts
import { Task } from "../../../examples/_harness/generated/ts/Task.generated";

// Fetch just the first match (with an optional sort). Resolves to null when
// nothing matches — no PaginatedResult wrapper.
export async function queryOne() {
  // #region example
  const topTask = await Task.queryOne(
    { completed: false },
    { sort: { priority: -1 } },
  );
  // #endregion example
  return topTask;
}
